agenta-cli/agenta/sdk/decorators/llm_entrypoint.py

```diff
@@ -296,6 +301,13 @@ async def execute_function(self, func: Callable[..., Any], *args, **func_params)
                 result = func(*args, **func_params["params"])
 
             if token is not None:
+                if WAIT_FOR_SPANS:
+                    remaining_steps = NOFSTEPS
+
+                    while not ag.tracing.is_trace_ready() and remaining_steps > 0:
+                        await asyncio.sleep(0.01)
+                        remaining_steps -= 1
```

Comment:
Is the user aware that the LLM response is waiting for the spans to be ready before showing it to them? If not, is there a way to make this configurable, maybe using env variables?

Reply:
Initially `WAIT_FOR_SPANS` was a flag, set to `False` for non-playground endpoints like `/generate_deployed`.
However, the issue is that in our SDK `flush_spans()` also needs to wait for all the spans to be ready. So it's not just for `inline traces` from `dump_trace()`.

Reply:
Ah, got it. Thank you for clarifying.